app/front/scripts/directives/export-control.js

```diff
@@ -0,0 +1,88 @@
+;(function(angular) {
+
+  var components = require('components');
+
+  var app = angular.module('Application');
+
+  function processCell(rows, rowIndex, colspan, rowspan, text) {
+    var colIndex = 0;
+    while (rows[rowIndex][colIndex] !== undefined) {
```

Comment:
Why do we need this loop? can't we get it using `length` or calculate it in the external loop?